src/test/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilderUnitTest.java

```diff
@@ -123,6 +122,26 @@ public void testSetOutputFile() throws IOException {
     }
 
     @Test
+    public void testDetermineOutputType() {
+        Assert.assertEquals(OutputType.VCF, VariantContextWriterBuilder.determineOutputTypeFromFile(this.vcf));
+        Assert.assertEquals(OutputType.BCF, VariantContextWriterBuilder.determineOutputTypeFromFile(this.bcf));
+        Assert.assertEquals(OutputType.VCF_STREAM, VariantContextWriterBuilder.determineOutputTypeFromFile(new File("/dev/stdout")));
```

Comment:
I've added a test for `.vcf.gz`.  What I'd mentioned though was using redirects.  This was previously unsupported, and I'm not even really sure how to test it as I'd have to fork another process and use the shell to redirect it's output stream in order to test it.  I don't think that's worth the effort.